src/persist/src/postgres.rs

```diff
@@ -246,8 +267,10 @@ impl Consensus for PostgresConsensus {
         let q = "SELECT sequence_number, data FROM consensus
              WHERE shard = $1 AND sequence_number >= $2
              ORDER BY sequence_number";
-        let client = self.client.lock().await;
-        let rows = client.query(&*q, &[&key, &from]).await?;
+        let mut client = self.client.lock().await;
+        let tx = self.tx(&mut client).await?;
+        let rows = tx.query(&*q, &[&key, &from]).await?;
+        tx.commit().await?;
```

Comment:
interesting! I didn't know you could just hold onto `rows` and iterate it after the txn commit

Reply:
yeah my understanding from the pg docs is that you definitely shouldn't do anything with any reads unless you are sure the txn committed successfully